### PR TITLE
Feature/cdap 14601 add preview for distributed mode in app fabric

### PR DESCRIPTION
**Pull request by jayfjin**
Change bindings and configurations to allow preview to run in distributed mode

https://builds.cask.co/browse/CDAP-DUT6727-3

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1719,6 +1719,8 @@
               </goals>
               <configuration>
                 <excludes>
+                  <exclude>cdap-spi/**</exclude>
```
Comment (chtyim) [inline pom.xml]: Shouldn't change this. Those modules are deleted, you should just delete those directories.

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewHttpModule.java [hunk 2]
```diff
@@ -16,30 +16,65 @@
...
+    return new PrivateModule() {
+      @Override
+      protected void configure() {
+        bind(DatasetDefinitionRegistryFactory.class)
+          .to(DefaultDatasetDefinitionRegistryFactory.class).in(Scopes.SINGLETON);
+
+        bind(DatasetFramework.class)
+          .annotatedWith(Names.named(DataSetsModules.BASE_DATASET_FRAMEWORK))
+          .to(RemoteDatasetFramework.class);
+        bind(LogReader.class).to(FileLogReader.class);
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewHttpModule.java:54]: looks like this is the only difference, in which case the distributed module can extend the standalone one to remove duplicate code.

**Review by albertshau — COMMENTED** [review]
just one comment

File: cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewHttpModule.java (hunk 2 above)
Comment (jayfjin) [reply cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewHttpModule.java:54]: Do we want to keep the modules linked via the inmemory/standalone/distributed pattern? If so I think the tradeoff is worth it to have a bit of duplicated code

Comment (albertshau) [reply cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewHttpModule.java:54]: you can still consolidate the common code with an internal class.

Comment (jayfjin) [reply cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewHttpModule.java:54]: sg, done

**Review by albertshau — APPROVED** [review]
lgtm

**Pull request merged**